samples/bluetooth/alexa_gadget/src/gadgets/gadgets_profile.c

```diff
@@ -0,0 +1,1231 @@
+{
+	bool ret;
+	int err;
+
+	err = encoded_data_append(data, len, BUF_USER_CONTROL_STREAM);
+	if (err == -EBUSY) {
+		LOG_WRN("Buffer busy");
+		return false;
+	} else if (err == -ENOMEM) {
+		LOG_WRN("Buffer overflow");
+		ret = false;
+		goto cleanup;
+	}
+
+	if (more_data) {
+		/* Wait for last piece of data */
+		return true;
+	}
+
+	err = gadgets_command_respond(conn);
+	if (err) {
+		LOG_WRN("Failed to reply to command message");
+		ret = false;
+		goto cleanup;
+	}
+
+	ret = true;
+
+cleanup:
+	encoded_data_reset();
+
+	return ret;
+}
+
+static bool gadgets_alexa_stream_cb(struct bt_conn *conn,
+				  const uint8_t *const data, uint16_t len, bool more_data)
+{
+	bool ret;
+	int err;
+
+	err = encoded_data_append(data, len, BUF_USER_ALEXA_STREAM);
+	if (err == -EBUSY) {
+		LOG_WRN("Buffer busy");
+		return false;
+	} else if (err == -ENOMEM) {
+		LOG_WRN("Buffer overflow");
+		ret = false;
+		goto cleanup;
+	}
+
+	__ASSERT(err == 0, "Unhandled error");
+
+	if (more_data) {
+		/* Wait for last piece of data */
+		return true;
+	}
+
+	pb_istream_t stream_in =
+		pb_istream_from_buffer(protobuf_encoded, protobuf_offset);
+	bool success = pb_decode(
+		&stream_in,
+		&directive_DirectiveParserProto_msg,
+		&protobuf_decoded.directive_parser);
+	if (!success) {
+		LOG_WRN("Failed to decode alexa message");
+		ret = false;
+		goto cleanup;
+	}
+
+	const char *name = protobuf_decoded.directive_parser.directive.header.name;
+	const char *namespace =
+		protobuf_decoded.directive_parser.directive.header.namespace;
+
+	LOG_DBG("Directive name: %s", log_strdup(name));
+	LOG_DBG("Directive namespace: %s", log_strdup(namespace));
+
+	ret = false;
+
+	for (int i = 0; i < ARRAY_SIZE(namespace_handlers); ++i) {
+		if (strcmp(namespace, namespace_handlers[i].namespace) == 0) {
+			/* Reminder: */
+			/* protobuf_encoded is re-used by handlers for outoing messages */
+			err = namespace_handlers[i].handler(
+				conn, name, protobuf_encoded, protobuf_offset);
+			if (err) {
+				LOG_ERR("Unsupported %s / %s", log_strdup(name),
+					log_strdup(namespace));
+			} else {
+				ret = true;
+			}
+			break;
+		}
+	}
+
+cleanup:
+	encoded_data_reset();
+
+	return ret;
+}
+
+static int directive_handler_discovery(
+	struct bt_conn *conn, const char *name,
+	const uint8_t *data, uint16_t len)
+{
+	int err;
+
+	if (strcmp(name, "Discover") != 0) {
+		return -EINVAL;
+	}
+
+	/* Construct response to discovery directive from hardcoded template. */
+	/* Serial number and device token are device-unique values */
+	/* that must be updated at runtime. */
+
+	pb_ostream_t stream_out =
+		pb_ostream_from_buffer(protobuf_encoded, sizeof(protobuf_encoded));
+	bool success = pb_encode(
+		&stream_out,
+		&alexaDiscovery_DiscoverResponseEventProto_msg,
+		&discovery_response_template);
+	__ASSERT(success, "protobuf_encoded is too small");
+	if (!success) {
+		return -EINVAL;
+	}
+
+	/* Replace serial and token placeholders with actual values */
+	char *serial_ptr;
```

Comment:
According to our codding standard variables should be defined at the beginning of the scope.

Reply:
I was not aware of this. Other samples does not seem to adhere to this rule